diff -durN crosstool-ng-1.22.0.orig/scripts/config.guess crosstool-ng-1.22.0/scripts/config.guess
--- crosstool-ng-1.22.0.orig/scripts/config.guess	2015-11-18 13:29:15.000000000 +0800
+++ crosstool-ng-1.22.0/scripts/config.guess	2017-04-13 16:43:45.328125000 +0800
@@ -918,7 +918,10 @@
 	GUESS=$UNAME_MACHINE-pc-cygwin
 	;;
     *:MINGW64*:*)
-	GUESS=$UNAME_MACHINE-pc-mingw64
+	GUESS=x86_64-w64-mingw32
+	;;
+    *:MINGW32*:*)
+	GUESS=i686-w64-mingw32
 	;;
     *:MINGW*:*)
 	GUESS=$UNAME_MACHINE-pc-mingw32
diff -durN crosstool-ng-1.22.0.orig/scripts/config.sub crosstool-ng-1.22.0/scripts/config.sub
--- crosstool-ng-1.22.0.orig/scripts/config.sub	2015-11-18 13:29:15.000000000 +0800
+++ crosstool-ng-1.22.0/scripts/config.sub	2017-04-13 16:45:44.218750000 +0800
@@ -388,11 +388,11 @@
 				basic_os=sysv
 				;;
 			mingw64)
-				basic_machine=x86_64-pc
-				basic_os=mingw64
+				basic_machine=x86_64-w64
+				basic_os=-mingw32
 				;;
 			mingw32)
-				basic_machine=i686-pc
+				basic_machine=i686-w64
 				basic_os=mingw32
 				;;
 			mingw32ce)
